python/ray/data/dataset.py

```diff
@@ -389,6 +389,18 @@ def parse_filename(row: Dict[str, Any]) -> Dict[str, Any]:
             logical_plan = LogicalPlan(map_op)
         return Dataset(plan, logical_plan)
 
+    def _set_name(self, name: Optional[str]):
```

Comment:
i think we can indicate that the set method is intended to be called by the user
```suggestion
    def set_name(self, name: Optional[str]):
```

Reply:
we will keep as private for now, in case we need to change it later.